crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -115,7 +123,10 @@ impl Dex {
         let sell = tokens.reference_price(&order.sell.token);
         let score =
             solution::Score::RiskAdjusted(self.risk.success_probability(swap.gas, gas_price, 1));
-        let Some(solution) = swap.into_solution(order.clone(), gas_price, sell, score) else {
+        let Some(solution) = swap
+            .into_solution(order.clone(), gas_price, sell, score, &self.simulator)
+            .await
+        else {
             tracing::debug!("no solution for swap");
```

Comment:
Same here, would like to use simulated gas estimate for `success_probability` calculation.